code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,847 @@
+/*
+Warping extracts crossbreed
+put up a rune with bluespace effects, lots of those runes are fluff or act as a passive buff, others are just griefing tools
+*/
+
+/obj/item/slimecross/warping
+	name = "warped extract"
+	desc = "It just won't stay in place."
+	icon_state = "warping"
+	effect = "warping"
+	colour = "grey"
+	///what runes will be drawn depending on the crossbreed color
+	var/obj/effect/warped_rune/runepath
+	/// the number of "charge" a bluespace crossbreed start with
+	var/warp_charge = INFINITY
+	///time it takes to store the rune back into the crossbreed
+	var/storing_time = 5 SECONDS
+	///time it takes to draw the rune
+	var/drawing_time = 5 SECONDS
+	var/max_cooldown = 30 SECONDS
+	var/cooldown = 0
+
+/obj/effect/warped_rune
+	name = "warped rune"
+	desc = "An unstable rune born of the depths of bluespace"
+	icon = 'icons/obj/slimecrossing.dmi'
+	icon_state = "greyspace_rune"
+	move_resist = INFINITY //here to avoid the rune being moved since it only sets it's turf once when it's drawn. doesn't include admin fuckery.
+	anchored = TRUE
+	layer = MID_TURF_LAYER
+	resistance_flags = FIRE_PROOF
+	var/dir_sound = 'sound/effects/phasein.ogg'
+	var/activated = FALSE
+	///is only used for bluespace crystal erasing as of now
+	var/storing_time = 5 SECONDS
+	///Nearly all runes needs to know which turf they are on
+	var/turf/rune_turf
+	var/deleteme = TRUE
+
+/obj/item/slimecross/warping/examine()
+	. = ..()
+	. += "It has [warp_charge] charge left"
+
+///runes can also be deleted by bluespace crystals relatively fast as an alternative to cleaning them.
+/obj/effect/warped_rune/attackby(obj/item/used_item, mob/user)
+	. = ..()
+	if(!istype(used_item,/obj/item/stack/sheet/bluespace_crystal) && !istype(used_item,/obj/item/stack/ore/bluespace_crystal))
```

Comment:
is this meant to be an or, not an and?

Reply:
if it is neither of these, the proc ends.